test/extended/etcd/helpers/helpers.go

```diff
@@ -184,37 +184,36 @@ func recoverClusterToInitialStateIfNeeded(ctx context.Context, t TestingT, machi
 }
 
 func DeleteSingleMachine(ctx context.Context, t TestingT, machineClient machinev1beta1client.MachineInterface) (string, error) {
-	waitPollInterval := 15 * time.Second
+	waitPollInterval := 30 * time.Second
 	waitPollTimeout := 5 * time.Minute
-	t.Logf("Waiting up to %s to delete a machine", waitPollTimeout.String())
-
 	machineToDelete := ""
-	err := wait.Poll(waitPollInterval, waitPollTimeout, func() (bool, error) {
-		machineList, err := machineClient.List(ctx, metav1.ListOptions{LabelSelector: masterMachineLabelSelector})
-		if err != nil {
-			return isTransientAPIError(t, err)
-		}
 
-		// Machine names are suffixed with an index number (e.g "ci-op-xlbdrkvl-6a467-qcbkh-master-0")
-		// so we sort to pick the lowest index, e.g master-0 in this example
-		machineNames := []string{}
-		for _, m := range machineList.Items {
-			machineNames = append(machineNames, m.Name)
-		}
-		sort.Strings(machineNames)
-		machineToDelete = machineNames[0]
-		t.Logf("attempting to delete machine %q", machineToDelete)
+	machineList, err := machineClient.List(ctx, metav1.ListOptions{LabelSelector: masterMachineLabelSelector})
+	if err != nil {
+		return "", fmt.Errorf("error listing master machines: '%w'", err)
+	}
 
+	// Machine names are suffixed with an index number (e.g "ci-op-xlbdrkvl-6a467-qcbkh-master-0")
+	// so we sort to pick the lowest index, e.g master-0 in this example
+	machineNames := []string{}
+	for _, m := range machineList.Items {
+		machineNames = append(machineNames, m.Name)
+	}
+	sort.Strings(machineNames)
+	machineToDelete = machineNames[0]
+
+	t.Logf("Waiting up to %s to delete a machine", waitPollTimeout.String())
+
+	err = wait.Poll(waitPollInterval, waitPollTimeout, func() (bool, error) {
```

Comment:
This doesn't need to be wrapped in poll either, but this way is at least functionally correct.